sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -321,4 +323,5 @@ object JDBCOptions {
   val JDBC_CONNECTION_PROVIDER = newOption("connectionProvider")
   val JDBC_PREPARE_QUERY = newOption("prepareQuery")
   val JDBC_PREFER_TIMESTAMP_NTZ = newOption("preferTimestampNTZ")
+  val JDBC_HINT_STRING = newOption("hint")
```

Comment:
Is there no way to standardize the type of hints?

Reply:
Shall we make the option is available for each of the built-in JDBC dialects ?

Reply:
> Shall we make the option is available for each of the built-in JDBC dialects ?

Currently, not all underlying JDBC data sources support the `hints` feature, such as `derby`, `postgresql` (supported by `pg_hint_plan` unofficial). In addition, `SQLServer` and `DB2` do not support comments-style hints.

Reply:
Could you explain the `pg_hint_plan` in detail? I want know the usage.

Reply:
It is a `postgresql` extension, reference:

- https://www.postgresql.org/about/news/pg_hint_plan-v170-released-2924/
- https://github.com/ossc-db/pg_hint_plan